
Hey cool - patching this paricular line seems to become a full-time job ...
or a good candidate for a cron job ;-)

Otherwise negative movements (0xff, ...) will be interpreted as 255 (...)
and since it is above 7 assigned to 0 :-(((

  - Rene Rebe <rene@rocklinux.org>

--- xc/programs/Xserver/hw/xfree86/input/mouse/mouse.c.orig	2003-10-10 13:06:30.000000000 +0200
+++ xc/programs/Xserver/hw/xfree86/input/mouse/mouse.c	2003-12-06 16:41:17.000000000 +0100
@@ -1386,7 +1386,7 @@
 		      (pBuf[0] & 0x80) >> 3;        /* button 5 */
 	    dx = (pBuf[0] & 0x10) ?    pBuf[1]-256  :  pBuf[1];
 	    dy = (pBuf[0] & 0x20) ?  -(pBuf[2]-256) : -pBuf[2];
-	    dz = (char)(pBuf[3] | ((pBuf[3] & 0x08) ? 0xf8 : 0));
+	    dz = (signed char)(pBuf[3] | ((pBuf[3] & 0x08) ? 0xf8 : 0));
 	    if ((pBuf[3] & 0xf8) && ((pBuf[3] & 0xf8) != 0xf8)) {
 		if (pMse->autoProbe) {
 		    SetMouseProto(pMse, PROT_EXPPS2);
